### ISSUE TITLE
Doesn't do anything

### ISSUE THREAD

**Issue by OnkelTem** [narcello/findead#37]
Hi!

I tried this tool with my project, but it didn't bring anything useful:
```
(develop *+$% u=)$ findead src/
==================================================================================================================== Findead is looking for components... =====================================================================================================================

=================================================================================================================================== Results ===================================================================================================================================
========================================================================================================================= No unused components found ==========================================================================================================================
================================================================== 1 browsed files in find: paths must precede expression: `jira.js'
find: possible unquoted pattern after predicate `-name'?
0,003 seconds ===================================================================
```

**Comment by narcello** [comment]
hmm.... strange.
try to run findead from root, like:
`findead ~/<your-project>/src`
or
`findead <your-project>/src`

**Comment by OnkelTem** [comment]
I stand in the root of the project.

```
(develop *+$% u=)$ ls -la
total 776
drwxrwxr-x   13 user user   4096 фев 17 20:42 .
drwxrwxr-x    6 user user   4096 янв 29 15:35 ..
-rw-rw-r--    1 user user   1638 фев 14 01:53 .babelrc.js
drwxrwxr-x    5 user user   4096 фев 17 16:14 config
drwxrwxr-x    6 user user   4096 янв 15 13:11 cypress
-rw-rw-r--    1 user user    674 янв 15 13:11 cypress.json
drwxrwxr-x    4 user user   4096 фев 14 15:54 dev
drwxrwxr-x    3 user user   4096 янв 21 17:14 dist
-rw-rw-r--    1 user user    414 янв 15 13:11 .editorconfig
drwxrwxr-x    2 user user   4096 янв 15 13:11 env-setup
drwxrwxr-x    8 user user   4096 фев 17 19:01 .git
-rw-rw-r--    1 user user     56 фев 17 16:14 .gitignore
-rw-rw-r--    1 user user   3801 фев  7 18:27 gulpfile.babel.js
drwxrwxr-x    6 user user   4096 фев 14 01:53 .idea
-rw-rw-r--    1 user user   4035 фев 17 16:19 Jenkinsfile
-rw-rw-r--    1 user user   1675 фев 17 16:14 jira.js
-rw-rw-r--    1 user user    430 янв 15 13:11 .jsbeautifyrc
-rw-rw-r--    1 user user     52 янв 15 13:11 LICENSE
drwxrwxr-x 1204 user user  45056 фев 12 21:41 node_modules
-rw-rw-r--    1 user user   7860 фев 17 16:14 package.json
-rw-rw-r--    1 user user 597829 фев 14 01:54 package-lock.json
-rw-rw-r--    1 user user    334 янв 15 13:11 .prettierrc
-rw-rw-r--    1 user user   2622 янв 15 13:11 README.md
-rw-rw-r--    1 user user    581 янв 15 13:11 spawn.js
drwxrwxr-x    6 user user   4096 янв 27 13:16 src
-rw-rw-r--    1 user user   1757 янв 15 13:11 svn.js
drwxrwxr-x    2 user user   4096 фев 17 16:14 .vscode
drwxrwxr-x    6 user user   4096 янв 15 13:11 web_modules

20:42:16 user:/projects/company/project/frontend
(develop *+$% u=)$ findead -r ~/src
Results
No unused components found
1 browsed files in find: paths must precede expression: `jira.js'
find: possible unquoted pattern after predicate `-name'?
0,005 seconds

20:42:30 user:/projects/company/project/frontend
(develop *+$% u=)$
```
Is there a sorta dev mode to debug the issue?

**Comment by narcello** [comment]
I think this not gone work:
`findead -r ~/src`

You should first exit from your project folder, maybe run `cd ..`, then you run:
`findead your-project-path`

today there's no have dev mode to debug :/
Anyway, the bug exists and I'll work on this.

Tell me please `findead` version and your SO. I'll let you know about the progress of resolution.
Thanks for open issue and describe de issue :+1:

**Comment by OnkelTem** [comment]
Wow wow, it has started working.
Waiting...

**Comment by OnkelTem** [comment]
Ok, it's finished working. 
First, it's included all the directories, including experiments from `dev` and modules from `web_modules`. But I could run it like this:
```
$ findead frontend/src
```
and it's created a report! Very good, it's found 27 dead components. But it's not that clear, because the reason I started to look up for tool like `findead` was that I came across a component which was not used, and the report doesn't include it. How can I figure out why?

**Comment by OnkelTem** [comment]
Well, despite the code is covered by NDI I think revealing one small and not that smart part is fine.
This is the component, missed from the report:
```
import _ from 'lodash';
import * as React from 'react';
import PropTypes from 'prop-types';
import ListItem from 'pages/common/react-md-components/Lists/ListItem';

import styles from './style';

class ScheduleItem extends React.PureComponent {
    constructor(props) {
        super(props);
    }

    render() {
        const { disabled, item, isSelected, onSelect } = this.props;
        return <ListItem primaryText="Recipes" />;
    }
}

ScheduleItem.propTypes = {
    disabled: PropTypes.bool,
    item: PropTypes.object,
    isSelected: PropTypes.bool,
    onSelect: PropTypes.func
};

export default ScheduleItem;

```
Location: `/src/pages/probes/services/item.jsx`

ideas?

**Comment by narcello** [comment]
I dont know what happens, I simulate this case in my computer, with `ScheduleItem` comp, same path, run findead from root and inside project, always show `ScheduleItem` in dead component report.


![image](https://user-images.githubusercontent.com/6786382/74683311-422f3600-51a7-11ea-835b-f88bdc78b191.png)

**Comment by OnkelTem** [comment]
> If you run at shell inside project:
> `grep '<ScheduleItem'> `
> do you get some results?

It's not clear, what do you mean?

Also, there are problems with the current report of 27 unused components: I've checked manually the first two components from the list and they're actually used, not dead at all :-/

**Comment by narcello** [comment]
Oh, now i understand, sorry. 
When you run `findead frontend/src`, the script will looking for components and there usages in `frontend/src`, if component was on this path and it usage outside `src/` it will be show in results like a dead component.

About the two components that you said, are they used inside frontend/src?

**Comment by OnkelTem** [comment]
> About the two components that you said, are they used inside frontend/src?

Sure.
Nevermind, this is the limitation of the selected approach. Yeah, I finally looked into the code and discovered a bash script! One doesn't simply take bash and except it to traverse JS internal structure :) Honestly, it's just no the right tool. Yet I admit it can help in some cases.

**Comment by narcello** [comment]
ok, can you send me prints and describe no unused component that findead report like unused component?
* path of declaration with print of component
* path of usage and print of usage with import

Just for me reproduce locally and solve this case.

**Comment by OnkelTem** [comment]
The code is covered by NDA, so I cannot really show a real example.

Have a look at this topic to get the idea of what bash script would barely do anyway:
https://github.com/davidmfoley/node-trucker/issues/25

Maybe this was the reason why your tool falsely displayed those files.

**Comment by narcello** [comment]
Ok, sure.

The findead is only for show unused components inside path that you pass. 

I'll study the case and work on this. Thanks for your contribution.

**Comment by narcello** [comment]
Hi @OnkelTem, 

1. I dont know yet why `findead` show your `ScheduleItem` component in dead component report.
2.  This above error, it is already solved on `v1.1.3`.

> Hi!
> 
> I tried this tool with my project, but it didn't bring anything useful:
> 
> ```
> (develop *+$% u=)$ findead src/
> ==================================================================================================================== Findead is looking for components... =====================================================================================================================
> 
> =================================================================================================================================== Results ===================================================================================================================================
> ========================================================================================================================= No unused components found ==========================================================================================================================
> ================================================================== 1 browsed files in find: paths must precede expression: `jira.js'
> find: possible unquoted pattern after predicate `-name'?
> 0,003 seconds ===================================================================
> ```
